src/core/QCheck2.mli

```diff
@@ -0,0 +1,2170 @@
+
+  val unit : unit t
+  (** [unit] is a printer of unit.
+
+      @since 0.6
+  *)
+
+  val int : int t
+  (** [int] is a printer of integer. *)
+
+  val bool : bool t
+  (** [bool] is a printer of boolean. *)
+
+  val float : float t
+  (** [float] is a printer of float. *)
+
+  val char : char t
+  (** [char] is a printer of character. *)
+
+  val string : string t
+  (** [string] is a printer of string. *)
+
+  val option : 'a t -> 'a option t
+  (** [option p] is a printer of ['a option], using [p] if it is a [Some]. *)
+
+  val pair : 'a t -> 'b t -> ('a*'b) t
+  (** [pair p1 p2] is a printer of pair. *)
+
+  val triple : 'a t -> 'b t -> 'c t -> ('a*'b*'c) t
+  (** [triple p1 p2 p3] is a printer of triple. *)
+
+  val quad : 'a t -> 'b t -> 'c t -> 'd t -> ('a*'b*'c*'d) t
+  (** [quad p1 p2 p3 p4] is a printer of quadruple. *)
+
+  val list : 'a t -> 'a list t
+  (** [list p] is a printer of list, using [p] for each element. *)
+
+  val array : 'a t -> 'a array t
+  (** [array p] is a printer of array, using [p] for each element. *)
+
+  val contramap : ('b -> 'a) -> 'a t -> 'b t
+  (** [contramap f p] transforms printer [p] into another using [f].
+
+      Note the reverse order of types in [f] which may be
+      conter-intuitive: indeed a function that {i prints} values of type
+      ['b] can be obtained by transforming a value of type ['b] to
+      ['a] using [f], and then by {i printing} this value of type ['a] using [p].
+  *)
+
+  val comap : ('b -> 'a) -> 'a t -> 'b t
+  (** @deprecated use {!contramap} instead. *)
+end
+
+(** Shrinking helper functions. *)
+module Shrink : sig
+  (** Shrinking is used to reduce the size of a counter-example. It tries
+      to make the counter-example smaller by decreasing it, or removing
+      elements, until the property to test holds again; then it returns the
+      smallest value that still made the test fail.
+
+      This is meant to help developers find a simpler counter-example to
+      ease investigation and find more easily the root cause (be it in the
+      tested code or in the test).
+
+      This module exposes helper functions that one can reuse in combination
+      with {!Gen.make_primitive} to craft custom primitive generators (not
+      by composing other generators). The vast majority of use cases will
+      probably not need this module.
+  *)
+
+  (** Util module representing a number type, used for ad hoc polymorphism of
+      some functions like {!number_towards}. *)
+  module type Number = sig
+    type t
+    val equal : t -> t -> bool
+    val div : t -> t -> t
+    val add : t -> t -> t
+    val sub : t -> t -> t
+    val of_int : int -> t
+  end
+
+  val number_towards : (module Number with type t = 'a) -> destination : 'a -> 'a -> 'a Seq.t
+  (** Shrink a number by edging towards a destination.
+
+      The destination is always the first value for optimal shrinking.
+
+      {[
+        let int64_towards_list destination x = List.of_seq @@
```

Comment:
seems like this kind of comment could be added to the very new test suite :)

Reply:
That's a good idea, done in 9c70e22ac63a67c23c66aadef86288d224cfd9a6 (took the liberty to reorganize a bit the test file, hope you don't mind; I find it clearer that way)